runtime/runtime.go

```diff
@@ -508,6 +508,15 @@ func (r *interpreterRuntime) parseAndCheckProgram(
 				sema.WithPredeclaredValues(valueDeclarations),
 				sema.WithPredeclaredTypes(typeDeclarations),
 				sema.WithValidTopLevelDeclarationsHandler(validTopLevelDeclarations),
+				sema.WithImportHandler(func(location ast.Location) sema.Import {
+					switch location {
+					case stdlib.CryptoChecker.Location:
+						return sema.CheckerImport{
+							Checker: stdlib.CryptoChecker,
+						}
+					}
+					return nil
```

Comment:
So if this import handler returns `nil`, does the checker call up to the host environment?

Reply:
It's actually the other way round: The checker first assumes the import is a program. If it is not, the import handler (if any) is called. If the import handler is still returning nil, then the import can not be resolved and an error is reported.

See https://github.com/onflow/cadence/blob/236a6ecdfbdbe7247bc3d6884209324fc0fe1e4d/runtime/sema/check_import_declaration.go#L55-L99

Reply:
Maybe I'm missing something, but doesn't this block contracts from being imported?

Reply:
Hmm, how would it block that?

Is it because there's no call to the runtime interface to get the code for the location?
If so, `ast.Program.ImportedPrograms` is resolved in the runtime before: 

https://github.com/onflow/cadence/blob/a968f040503e86274f7ac9c522110ce66e99b110/runtime/runtime.go#L434-L435

Reply:
Oh I see, that makes sense!